helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -56,6 +57,24 @@
 import org.slf4j.LoggerFactory;
 
 public class ZkBaseDataAccessor<T> implements BaseDataAccessor<T> {
+
+  // Designates which mode ZkBaseDataAccessor should be created in. If not specified, it will be
+  // created on SHARED mode.
+  enum ZkClientType {
```

Comment:
public enum? Otherwise, how client get access to the enum in the constructor? 
I think for the ease of client, the builder pattern is more straightforward to read. Otherwise, one needs to read all the comments of enums and NOTE sessions of each constructor to figure out the right one.
`ZkBaseDataAccessor.Builder(zkAddress).setSerializer(XXX).buildDedicate()`
`ZkBaseDataAccessor.Builder(zkAddress).setSerializer(XXX).buildShared()`